presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1479,4 +1480,44 @@ public ActualProperties getProperties()
             return properties;
         }
     }
+
+    private static boolean isPartitioningCompatible(List<PartitioningHandle> partitioningHandles)
```

Comment:
nit:  `compatible partitioning` is now a bit vague term now , since there can be connector compatible partitioning as well.  cc @shixuan-fan 

I don't have a better suggestion though. What about `sameOrSystemCompatiblePartitions`? cc @rschlussel